server/src/main/java/org/elasticsearch/index/query/RankDocsQueryBuilder.java

```diff
@@ -45,9 +47,11 @@ public RankDocsQueryBuilder(StreamInput in) throws IOException {
         if (in.getTransportVersion().onOrAfter(TransportVersions.V_8_16_0)) {
             this.queryBuilders = in.readOptionalArray(c -> c.readNamedWriteable(QueryBuilder.class), QueryBuilder[]::new);
             this.onlyRankDocs = in.readBoolean();
+            this.minScore = in.readFloat();
```

Comment:
As this is written, it will break transport serialization. 

What you want to do, is register a new `TransportVersion` for your change in `TransportVersions.java`. You'll then reference this new transport version to serialize the min score - older transport versions will always just use a default value.